packages/flutter_tools/lib/src/android/deferred_components.dart

```diff
@@ -0,0 +1,674 @@
+''', flush: false);
+
+      for (String line in requiredLines) {
+        stringResOutput.writeAsStringSync('    $line', flush: false, mode: FileMode.append);
+      }
+      stringResOutput.writeAsStringSync(
+'''
+    <!--End Deferred Components strings.-->
+</resources>
+
+''', flush: true, mode: FileMode.append);
+      modifiedFiles.add(stringResOutput.path);
+    }
+    return modifiedFiles;
+  }
+
+  Map<String, dynamic> _checkLoadingUnitGolden(List<LoadingUnit> generatedLoadingUnits, Map<String, dynamic> deferredComponentsGolden) {
+    List<LoadingUnit> goldenLoadingUnits = deferredComponentsGolden['loading-units'] ?? [];
+    Set<LoadingUnit> unmatchedLoadingUnits = {};
+    unmatchedLoadingUnits.addAll(goldenLoadingUnits);
+    List<LoadingUnit> newLoadingUnits = [];
+    Set<int> addedNewIds = {};
+    for (LoadingUnit genUnit in generatedLoadingUnits) {
+      bool matched = false;
+      for (LoadingUnit goldUnit in goldenLoadingUnits) {
+        if (genUnit.equalsIgnorePath(goldUnit)) {
+          matched = true;
+          unmatchedLoadingUnits.remove(goldUnit);
+          break;
+        }
+      }
+      if (!matched && !addedNewIds.contains(genUnit.id)) {
+        newLoadingUnits.add(genUnit);
+        addedNewIds.add(genUnit.id);
+      }
+    }
+    Map<String, dynamic> results = {};
+    results['new'] = newLoadingUnits;
+    results['missing'] = unmatchedLoadingUnits;
+    results['match'] = newLoadingUnits.isEmpty && unmatchedLoadingUnits.isEmpty;
+    return results;
+  }
+
+  Map<String, dynamic> _parseGolden(File goldenFile) {
+    if (!goldenFile.existsSync()) {
+      return <String, dynamic>{};
+    }
+    var data = loadYaml(goldenFile.readAsStringSync());
+    Map<String, dynamic> output = <String, dynamic>{};
+
+    List<LoadingUnit> loadingUnits = <LoadingUnit>[];
+    output['loading-units'] = loadingUnits;
+    if (data.containsKey('loading-units') && data['loading-units'] != null) {
+      for (var loadingUnitData in data['loading-units']) {
+        List<String> libraries = <String>[for (String lib in loadingUnitData['libraries']) lib];
+        loadingUnits.add(
+            LoadingUnit(
+              id: loadingUnitData['id'],
+              path: null,
+              libraries: libraries,
+            ));
+      }
+    }
+    return output;
+  }
+
+  void _writeGolden(List<LoadingUnit> generatedLoadingUnits) {
+    File goldenFile = env.projectDir.childFile(_kDeferredComponentsGoldenFileName);
+    if (goldenFile.existsSync()) {
+      goldenFile.deleteSync();
+    }
+    goldenFile.createSync(recursive: true);
+
+    goldenFile.writeAsStringSync(
+  '''
+# ====================================================================
+# The contents of this file are automatically generated and it is not
+# recommended to modify this file manually.
+# ====================================================================
+#
+# This list of loading units are a snapshot of the loading units generated
```

Comment:
describe the problem statement first. i.e. why is there a golden file and what is it solving